prusti-viper/src/encoder/encoder.rs

```diff
@@ -229,8 +231,9 @@ impl<'v, 'tcx> Encoder<'v, 'tcx> {
         self.error_manager.borrow_mut()
     }
 
-    pub fn get_viper_program(&self) -> vir::Program {
+    pub fn finalize_viper_program(&self, name: String) -> vir::Program {
         vir::Program {
+            name,
             domains: self.get_used_viper_domains(),
             fields: self.get_used_viper_fields(),
             builtin_methods: self.get_used_builtin_methods(),
```

Comment:
Is this going to add more and more (potentially unused) domains and fields to each program as we progress through the processing queue?

Reply:
I have created this PR just to see whether the test suite goes through. I will fix that eventually.